hpat/datatypes/hpat_pandas_series_functions.py

```diff
@@ -534,52 +534,42 @@ def hpat_pandas_series_copy_impl(self, deep=True):
-#        Test: python -m hpat.runtests hpat.tests.test_series.TestSeries.test_series_head_index3
-#        Test: python -m hpat.runtests hpat.tests.test_series.TestSeries.test_series_head_index4
-#        Test: python -m hpat.runtests hpat.tests.test_series.TestSeries.test_series_head_parallel1
-#        Test: python -m hpat.runtests hpat.tests.test_series.TestSeries.test_series_head_index_parallel1
-#        Test: python -m hpat.runtests hpat.tests.test_series.TestSeries.test_series_head_index_parallel2
-#        Test: python -m hpat.runtests hpat.tests.test_series.TestSeries.test_series_head_noidx
-#        Test: python -m hpat.runtests hpat.tests.test_series.TestSeries.test_series_head_idx
-#
-#     Parameters
-#     -----------
-#     n: :obj:`int`
-#                input argument, default 5
-#     Returns
-#     -------
-#     :obj:`pandas.Series`
-#          returns The first n rows of the caller object.
-#     """
-#
-#     _func_name = 'Method head().'
-#
-#     if not isinstance(self, SeriesType):
-#         raise TypingError('{} The object must be a pandas.series. Given: {}'.format(_func_name, self))
-#
-#     if not isinstance(n, (types.Integer, types.Omitted)) and n != 5:
-#         raise TypingError('{} The parameter must be an integer type. Given type n: {}'.format(_func_name, n))
-#
-#     if isinstance(self.index, types.NoneType):
-#         def hpat_pandas_series_head_impl(self, n=5):
-#             return pandas.Series(self._data[:n])
-#
-#         return hpat_pandas_series_head_impl
-#     else:
-#         def hpat_pandas_series_head_index_impl(self, n=5):
-#             return pandas.Series(self._data[:n], self._index[:n])
-#
-#         return hpat_pandas_series_head_index_impl
+@overload_method(SeriesType, 'head')
+def hpat_pandas_series_head(self, n=5):
+    """
+    Pandas Series method :meth:`pandas.Series.head` implementation.
+
+    .. only:: developer
+       Test: python -m -k hpat.runtests hpat.tests.test_series.TestSeries.test_series_head*
+
+    Parameters
+    -----------
+    n: :obj:`int`, default 5
+        input argument, default 5
+    Returns
+    -------
+    :obj:`pandas.Series`
+        returns: The first n rows of the caller object.
+    """
+
+    _func_name = 'Method head().'
+
+    if not isinstance(self, SeriesType):
+        raise TypingError('{} The object must be a pandas.series. Given: {}'.format(_func_name, self))
+
+    if not isinstance(n, (types.Integer, types.Omitted)) and n != 5:
+        raise TypingError('{} The parameter must be an integer type. Given type n: {}'.format(_func_name, n))
+
+    if isinstance(self.index, types.NoneType):
+        def hpat_pandas_series_head_impl(self, n=5):
+            return pandas.Series(self._data[:n])
+
+        return hpat_pandas_series_head_impl
+    else:
+        def hpat_pandas_series_head_index_impl(self, n=5):
+            return pandas.Series(self._data[:n], self._index[:n])
```

Comment:
Could you construct output Series passing parameter `name` as `self._name`? It should be supported.